growattServer/v1_api.py

```
@@ -0,0 +1,605 @@
+        """
+        Read Time-of-Use (TOU) settings from a Growatt MIN/TLX inverter.
+
+        Retrieves all 9 time segments from a Growatt MIN/TLX inverter and
+        parses them into a structured format.
+
+        Note that this function uses min_settings() internally to get the data, so same endpoint rate limit applies
+
+        Args:
+            device_sn (str): The device serial number of the inverter
+            settings_data (dict, optional): Settings data from min_settings call to avoid repeated API calls.
+                                        Can be either the complete response or just the data portion.
+
+        Returns:
+            list: A list of dictionaries, each containing details for one time segment:
+                - segment_id (int): The segment number (1-9)
+                - batt_mode (int): 0=Load First, 1=Battery First, 2=Grid First
+                - mode_name (str): String representation of the mode
+                - start_time (str): Start time in format "HH:MM"
+                - end_time (str): End time in format "HH:MM"
+                - enabled (bool): Whether the segment is enabled
+
+        Example:
+            api = GrowattApi(token="your_api_token")
+
+            # Option 1: Make a single call
+            tou_settings = api.min_read_tou_settings("DEVICE_SERIAL_NUMBER")
+
+            # Option 2: Reuse existing settings data
+            settings_response = api.min_settings("DEVICE_SERIAL_NUMBER")
+            tou_settings = api.min_read_tou_settings("DEVICE_SERIAL_NUMBER", settings_response)
+
+        """
+
+        # Process the settings data
+        if settings_data is None:
+            # Fetch settings if not provided
+            settings_response = self.min_settings(device_sn=device_sn)
+            if settings_response.get('error_code', 1) != 0:
+                print(
+                    f"Failed to get settings, error: {settings_response.get('error_msg', 'Unknown error')}")
```

I think we'll want to throw these as errors so they can be caught by the code implementing this function.